### PR TITLE
Short message sending

### PR DESCRIPTION
**Pull request by mbonani**
Enabling a def for short message sending specially for Thymio RF

### CODE REVIEW

File: transport/buffer/vm-buffer.c [hunk 1]
```diff
@@ -97,15 +97,36 @@ void AsebaSendMessageWords(AsebaVMState *vm, uint16 type, const uint16* data, ui
 
 void AsebaSendVariables(AsebaVMState *vm, uint16 start, uint16 length)
 {
-	uint16 i;
 
-	buffer_pos = 0;
+	uint16 i;
+#ifndef ASEBA_LIMITED_MESSAGE_SIZE 
+        buffer_pos = 0;
```
Comment (stephanemagnenat) [inline transport/buffer/vm-buffer.c]: There are spaces instead of tabs here, please correct ;-)

File: transport/buffer/vm-buffer.c [hunk 2]
```diff
@@ -97,15 +97,36 @@ void AsebaSendMessageWords(AsebaVMState *vm, uint16 type, const uint16* data, ui
...
+        AsebaSendBuffer(vm, buffer, buffer_pos);
 	buffer_add_uint16(ASEBA_MESSAGE_VARIABLES);
 	buffer_add_uint16(start);
 	for (i = start; i < start + length; i++)
 		buffer_add_uint16(vm->variables[i]);
+        AsebaSendBuffer(vm, buffer, buffer_pos);
+#else
+        const uint16 MAX_VARIABLES_SIZE = ((100 - 6)/2);
```
Comment (stephanemagnenat) [inline transport/buffer/vm-buffer.c]: There are spaces instead of tabs here, please correct

File: transport/buffer/vm-buffer.c [hunk 3]
```diff
@@ -98,14 +98,34 @@ void AsebaSendMessageWords(AsebaVMState *vm, uint16 type, const uint16* data, ui
 void AsebaSendVariables(AsebaVMState *vm, uint16 start, uint16 length)
 {
 	uint16 i;
-
+#ifndef ASEBA_LIMITED_MESSAGE_SIZE 
 	buffer_pos = 0;
+	AsebaSendBuffer(vm, buffer, buffer_pos);
```
Comment (stephanemagnenat) [inline transport/buffer/vm-buffer.c]: why that line?

File: transport/buffer/vm-buffer.c [hunk 4]
```diff
@@ -98,14 +98,34 @@ void AsebaSendMessageWords(AsebaVMState *vm, uint16 type, const uint16* data, ui
 void AsebaSendVariables(AsebaVMState *vm, uint16 start, uint16 length)
 {
 	uint16 i;
-
+#ifndef ASEBA_LIMITED_MESSAGE_SIZE
```
Comment (stephanemagnenat) [inline transport/buffer/vm-buffer.c]: please add a comment saying in which context this is useful

File: transport/buffer/vm-buffer.c [hunk 5]
```diff
@@ -98,14 +98,34 @@ void AsebaSendMessageWords(AsebaVMState *vm, uint16 type, const uint16* data, ui
...
-
 	AsebaSendBuffer(vm, buffer, buffer_pos);
+#else
+	const uint16 MAX_VARIABLES_SIZE = ((100 - 6)/2);
+	do {
+		uint16 size;
+		buffer_pos = 0;
+		buffer_add_uint16(ASEBA_MESSAGE_VARIABLES);
+		buffer_add_uint16(start);
+		if(length > MAX_VARIABLES_SIZE)
```
Comment (stephanemagnenat) [inline transport/buffer/vm-buffer.c]: put space before (

**Comment by mbonani** [comment]
will remake a pull request from a new commit

**Pull request closed without merging**